grpc-ingest/src/grpc.rs

```diff
@@ -119,97 +131,140 @@ pub async fn run(config: ConfigGrpc) -> anyhow::Result<()> {
     let config = Arc::new(config);
     let connection = redis_client.get_multiplexed_tokio_connection().await?;
 
-    let mut shutdown = create_shutdown()?;
-
     let pipe = Arc::new(Mutex::new(TrackedPipeline::default()));
 
-    let mut accounts = HashMap::with_capacity(1);
-    let mut transactions = HashMap::with_capacity(1);
-
-    accounts.insert("das".to_string(), config.accounts.filter.clone().to_proto());
-    transactions.insert(
-        "das".to_string(),
-        config.transactions.filter.clone().to_proto(),
-    );
-
-    let request = SubscribeRequest {
-        accounts,
-        transactions,
-        ..Default::default()
-    };
-
-    let mut dragon_mouth_client =
-        GeyserGrpcClient::build_from_shared(config.geyser_endpoint.clone())?
-            .x_token(config.x_token.clone())?
-            .connect_timeout(Duration::from_secs(10))
-            .timeout(Duration::from_secs(10))
-            .connect()
-            .await
-            .context("failed to connect to gRPC")?;
-
-    let (mut subscribe_tx, stream) = dragon_mouth_client
-        .subscribe_with_request(Some(request))
-        .await?;
-
-    tokio::pin!(stream);
-
-    let exec = Executor::builder(Nonblock(Tokio))
-        .max_concurrency(Some(config.max_concurrency))
-        .build_async(GrpcJobHandler {
-            config: Arc::clone(&config),
-            connection: connection.clone(),
-            pipe: Arc::clone(&pipe),
-        })?;
-
-    let deadline_config = Arc::clone(&config);
-
-    loop {
-        tokio::select! {
-            _ = sleep(deadline_config.redis.pipeline_max_idle) => {
-                exec.push(GrpcJob::FlushRedisPipe);
-            }
-            Some(Ok(msg)) = stream.next() => {
-                match msg.update_oneof {
-                    Some(UpdateOneof::Account(_)) | Some(UpdateOneof::Transaction(_)) => {
-                        exec.push(GrpcJob::ProcessSubscribeUpdate(Box::new(msg)));
+    let subscriptions = config.subscriptions.clone();
+
+    let mut subscription_tasks: Vec<JoinHandle<Result<(), anyhow::Error>>> =
+        Vec::with_capacity(subscriptions.len());
+
+    for subscription in subscriptions {
+        let subscription_label = subscription.0.clone();
+        let subscription_config = Arc::new(subscription.1);
+        let config = Arc::clone(&config);
+        let connection = connection.clone();
+        let pipe = Arc::clone(&pipe);
+        let subscription_task_handle: JoinHandle<Result<(), anyhow::Error>> = tokio::spawn(
+            async move {
+                let mut shutdown = create_shutdown()?;
```

Comment:
Its likely best to have 1 shutdown signal for the main handler and work with that to signal shutdowns to other threads so you catch the shutdown just once.

Reply:
done , created a standalone shutdown mechanism